### PR TITLE
[Home Page Redesign] Refine responsiveness

### PR DESCRIPTION
**Pull request by ashleemboyer**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ambhome-page-responsiveness-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/0/1204200670480307) 🎟️

## 🗒️ What

- Makes a `size` fix for `StandaloneLinkContents` (details in self-review comment)
- Adjusts text sizes, spacing, etc. for breakpoints outlined in the design

## 📸 Design Screenshots

<details>
  <summary><code>1920px</code> wide</summary>
  <a href="https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=20982-279288&t=jpTcOHcViuKae76d-11">Figma link</a>
  <br />
  
</details>

<details>
  <summary><code>1440px</code> wide</summary>
  <a href="https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=20706-250693&t=jpTcOHcViuKae76d-11">Figma link</a>
  <br />
  
</details>

<details>
  <summary><code>999px</code> wide</summary>
  <a href="https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=20706-252403&t=jpTcOHcViuKae76d-11">Figma link</a>
  <br />
  
</details>

<details>
  <summary><code>1920px</code> wide</summary>
  <a href="https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=20706-253431&t=jpTcOHcViuKae76d-11">Figma link</a>
  <br />
  
</details>

## 🧪 Testing

- [ ] Ensure elements are sized, spaced, and placed as expected in the following viewport widths:
  - [ ] `1920px`
  - [ ] `1440px`
  - [ ] `1001px`
  - [ ] `999px`
  - [ ] `700px`
  - [ ] `500px`
  - [ ] `320px`
- [ ] Ensure the previous is still true with 200% text-only zoom

## 💭 Anything else?

- Further polish will come in the next PR!

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204200670480307

### CODE REVIEW

File: src/components/standalone-link/index.tsx
Comment (ashleemboyer) [inline src/components/standalone-link/index.tsx:1]: 🗒️ The changes in this file were needed after I noticed that `StandaloneLinkContents` was not dynamically sizing up when testing 200% text-only zoom.

File: src/views/homepage/components/chiclets/chiclets.module.css
Comment (ashleemboyer) [inline src/views/homepage/components/chiclets/chiclets.module.css:33]: 🗒️ Unfortunately can't use `composes` to make these shifts in media queries, so I had to pull out the tokens and apply them directly.

File: src/views/homepage/components/chiclets/index.tsx
Comment (ashleemboyer) [inline src/views/homepage/components/chiclets/index.tsx:20]: 🗒️ This text was updated in the designs, so updated here to reflect.

File: src/views/homepage/components/chiclets/index.tsx
Comment (ashleemboyer) [inline src/views/homepage/components/chiclets/index.tsx:22]: 🗒️ This class wasn't actually defined in the CSS, so removed it.

File: src/views/homepage/components/chiclets/index.tsx
Comment (ashleemboyer) [inline src/views/homepage/components/chiclets/index.tsx:38]: 🗒️ This text needed to be wrapped to apply styles to it. Didn't notice before that styles hadn't been applied already. 😊

File: src/views/homepage/components/featured-content-grid/certifications-featured-card.tsx
Comment (ashleemboyer) [inline src/views/homepage/components/featured-content-grid/certifications-featured-card.tsx:27]: 🗒️ I took out `Heading` and `Text` since the styles those apply (`font-size` in particular) change based on `@media` queries.

File: src/views/homepage/components/featured-content-grid/featured-content-grid.module.css
Comment (ashleemboyer) [inline src/views/homepage/components/featured-content-grid/featured-content-grid.module.css:22]: 🗒️ I found that these `@media` queries based on `rem` units help us be more responsive to text-only zoom!

File: src/views/homepage/components/featured-content-grid/search-featured-card.module.css
Comment (ashleemboyer) [inline src/views/homepage/components/featured-content-grid/search-featured-card.module.css:130]: TODO for me before merging

File: src/views/homepage/components/featured-content-grid/waf-featured-card.tsx
Comment (ashleemboyer) [inline src/views/homepage/components/featured-content-grid/waf-featured-card.tsx:17]: 🗒️ Same note here re: needing more control over changing styles like `font-size` with `@media` queries.

File: src/views/homepage/components/chiclets/chiclets.module.css
Comment (zchsh) [inline src/views/homepage/components/chiclets/chiclets.module.css:39]: ⛏️ I'm seeing the chiclets wrap on large viewports (at default text zoom level, in FireFox), which differs a bit from design:

2023-04-20-wrap-issue

I'm not sure what exact value is needed here (probably hard to predict based on text layout), but ~ 12px seems to do it.

```suggestion
	gap: 12px;
```

File: src/views/homepage/components/chiclets/chiclets.module.css
Comment (zchsh) [inline src/views/homepage/components/chiclets/chiclets.module.css:46]: ⛏️ Similar to the desktop wrap issue, we might need to bump this down to `7px` or something (or maybe some other change?) if we want the chiclets to stay on one line at specific viewport widths (`999px` shown below):

CleanShot 2023-04-20 at 10 38 51@2x

File: src/views/homepage/components/featured-content-grid/certifications-featured-card.module.css
Comment (zchsh) [inline src/views/homepage/components/featured-content-grid/certifications-featured-card.module.css:38]: ⛏️  I might be looking at the wrong designs, and also I feel like the `600px - 1000px` viewport range is just hard to design for sometimes, but I'm seeing a significant difference in font size so wanted to flag in case it's something we want to tweak:

CleanShot 2023-04-20 at 10 42 49@2x

**Review by zchsh — APPROVED** [review]
✅ Approving from a code perspective, everything looks good to me on that front.

⛏️ (🚧 ?) Running through everything across the viewports listed in `Testing Steps`, I did find a few design nitpicks, but will defer to design reviewers on whether any of them should be blocking.

⛏️ One other nitpick I noticed is some funky text wrapping in the `Search` card on mobile. I think removing some amount of spacing either within or around the card could improve readability a lot at our smallest supported viewport size of `320px`:

| Current | Padding removed as rough demo |
| - | - | 
| current | padding removed as rough demo |

**Comment by ashleemboyer** [comment]
@zchsh @kendallstrautman just a heads up that when I sync'd with @kellyschummer a little bit ago, she let me know she's OK with merging this PR and then resolving the polish & nits in the final pre-release PR that'll open for review tomorrow. 😊

**Pull request merged**
